public/components/trace_analytics/components/common/redirection_helpers.tsx

```diff
@@ -0,0 +1,183 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import dateMath from '@elastic/datemath';
+import moment from 'moment';
+import { DataSourceOption } from '../../../../../../../src/plugins/data_source_management/public/components/data_source_menu/types';
+import {
+  DEFAULT_DATA_SOURCE_NAME,
+  DEFAULT_DATA_SOURCE_TYPE,
+} from '../../../../../common/constants/data_sources';
+import {
+  observabilityLogsID,
+  observabilityTracesNewNavID,
+} from '../../../../../common/constants/shared';
+import { TRACE_ANALYTICS_DATE_FORMAT } from '../../../../../common/constants/trace_analytics';
+import { TraceAnalyticsMode } from '../../../../../common/types/trace_analytics';
+import { coreRefs } from '../../../../framework/core_refs';
+import { FilterType } from './filters/filters';
+import { TraceSettings } from './helper_functions';
+
+const redirectionToLogsApp = ({
+  startTime,
+  endTime,
+  dataSourceMDSId,
+  correlatedFieldName,
+  correlatedFieldValue,
+}: {
+  startTime: string;
+  endTime: string;
+  dataSourceMDSId: DataSourceOption[];
+  correlatedFieldName: string;
+  correlatedFieldValue: string;
+}) => {
+  const correlatedLogsIndex = TraceSettings.getCorrelatedLogsIndex();
+  const correlatedTimestampField = TraceSettings.getCorrelatedLogsFieldMappings().timestamp;
+  if (coreRefs?.dataSource?.dataSourceEnabled || coreRefs.chrome?.navGroup.getNavGroupEnabled()) {
+    coreRefs?.application!.navigateToApp('data-explorer', {
+      path: `discover#?_a=(discover:(columns:!(_source),isDirty:!f,sort:!()),metadata:(view:discover))&_g=(filters:!(),refreshInterval:(pause:!t,value:0),time:(from:'${startTime}',to:'${endTime}'))&_q=(filters:!(),query:(dataset:(dataSource:(id:'${
+        dataSourceMDSId[0].id ?? ''
+      }',title:'${dataSourceMDSId[0].label}',type:DATA_SOURCE),id:'${
+        dataSourceMDSId[0].id ?? ''
+      }::${correlatedLogsIndex}',timeFieldName:'${correlatedTimestampField}',title:'${correlatedLogsIndex}',type:INDEXES),language:PPL,query:'source%20%3D%20${correlatedLogsIndex}%20%7C%20where%20${correlatedFieldName}%20%3D%20%22${correlatedFieldValue}%22'))`,
```

Comment:
can the url encode be done after string manipulations of the query rather than hard coded? would logs index or field name/value contain unescaped characters

Reply:
makes sense updated this [3de427e](https://github.com/opensearch-project/dashboards-observability/pull/2398/commits/3de427e9fe08191d804ca0496fec4e718e398b32) to ensure the URL encode happens after variables are added in.